gcloud/bigquery/dataset.py

```diff
@@ -204,6 +273,24 @@ def _require_client(self, client):
             client = self._client
         return client
 
+    def _parse_access_grants(self, access):
+        """Parse a resource fragment into a set of access grants.
+
+        :type access: list of mappings
+        :param access: each mapping represents a single access grant
+
+        :rtype: list of :class:`AccessGrant`
+        :returns: a list of parsed grants
+        """
+        result = []
+        for grant in access:
+            grant = grant.copy()
+            role = grant.pop('role')
+            entity_type, entity_id = list(grant.items())[0]
```

Comment:
09c1d14 makes the code resilient against the hypothetical case that the back-end returns more than one entity type in a given role grant.